Assets/Scripts/BeamManager.cs

```diff
@@ -0,0 +1,144 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+public class BeamManager : MonoBehaviour
+{
+    private enum difficultyState
+    {
+        easy, medium, hard, expert
+    }
+
+    [SerializeField]
+    [Tooltip("amount of time in between each spawn")]
+    private float baseSpawnTime = 3;
+    private float spawnTime;
+
+    private float startingSpeed;
+    private float timer;
+    private float spawnHeight = -102;
+    private float lastSpawnTime;
+
+    [SerializeField]
+    private float mediumTime = 20f, hardTime = 50f, expertTime = 75f;
+    private difficultyState state;
+
+    [SerializeField]
+    private GameObject beamHorizontal, beamVertical;
+
+    // Start is called before the first frame update
+    void Start()
+    {
+        timer = 0;
+        lastSpawnTime = 0;
+        spawnTime = baseSpawnTime;
+        startingSpeed = BeamMovement.speed; //bee movie . speed
+        state = difficultyState.easy;
+    }
+
+    // Update is called once per frame
+    void Update()
+    {
+        timer += Time.deltaTime;
+        if (BeamMovement.speed < BeamMovement.maxSpeed)
+        {
+            BeamMovement.speed += Time.deltaTime / 2;
+        }
+        spawnTime = baseSpawnTime / (BeamMovement.speed / startingSpeed);
+
+        if (timer >= lastSpawnTime + spawnTime)
+        {
+            lastSpawnTime = timer;
+            SpawnRandomBeamLayer();
+        }
+
+        if (timer >= mediumTime && timer < hardTime)
+        {
+            Debug.Log("Medium");
+            state = difficultyState.medium;
+        }
+        else if (timer >= hardTime && timer < expertTime)
+        {
+            Debug.Log("hard");
+            state = difficultyState.hard;
+        }
+        else if (timer >= expertTime)
+        {
+            Debug.Log("expert");
+            state = difficultyState.expert;
+        }
+        else
+        {
+            Debug.Log("easy");
+            state = difficultyState.easy;
+        }
```

Comment:
Maybe this block of if else can be refactored into a SetDifficulty() method to improve clarity